homeassistant/components/aemet/weather.py

```diff
@@ -137,27 +143,31 @@ def __init__(
         super().__init__(coordinator)
         self._forecast_mode = forecast_mode
         self._attr_entity_registry_enabled_default = (
-            self._forecast_mode == FORECAST_MODE_DAILY
+            self._forecast_mode == AOD_FORECAST_DAILY
         )
         self._attr_name = name
         self._attr_unique_id = unique_id
 
     @property
     def condition(self):
         """Return the current condition."""
-        return self.coordinator.data[ATTR_API_CONDITION]
+        cond = self.get_aemet_value([AOD_WEATHER, AOD_CONDITION])
+        return CONDITIONS_MAP.get(cond)
 
     def _forecast(self, forecast_mode: str) -> list[Forecast]:
         """Return the forecast array."""
-        forecasts = self.coordinator.data[FORECAST_MODE_ATTR_API[forecast_mode]]
+        forecasts = self.get_aemet_value([AOD_TOWN, forecast_mode, AOD_FORECAST])
         forecast_map = FORECAST_MAP[forecast_mode]
-        return cast(
-            list[Forecast],
-            [
-                {ha_key: forecast[api_key] for api_key, ha_key in forecast_map.items()}
-                for forecast in forecasts
-            ],
-        )
+        forecast_list: list[dict[str, Any]] = []
+        for forecast in forecasts:
+            cur_forecast: dict[str, Any] = {}
+            for api_key, ha_key in forecast_map.items():
+                value = forecast[api_key]
+                if api_key == AOD_CONDITION:
+                    value = CONDITIONS_MAP.get(value)
+                cur_forecast[ha_key] = value
+            forecast_list += [cur_forecast]
+        return cast(list[Forecast], forecast_list)
```

Comment:
Is there a way to avoid building this every time its called? This looks like it will make it rather expensive to construct.

I need to dig to see how often this is run to see if it actually matters but github is rate limiting right now so I'll have to do it in a bit

Reply:
> Is there a way to avoid building this every time its called? This looks like it will make it rather expensive to construct.
> 
> I need to dig to see how often this is run to see if it actually matters but github is rate limiting right now so I'll have to do it in a bit

Would something like [this](https://github.com/Noltari/home-assistant-core/commit/7e480df3e7f6c8bd93af35ffec19c2d1d49c896a) be acceptable?
I know it's a bit hacky, but it's the only solution I can think of right now...

Reply:
Done in dd5078ad8fa5b2c1ecdbed96f8045ac314f8b188 with a better approach (and less hacky xD)